src/model/sample.rs

```diff
@@ -340,7 +349,8 @@ impl Sample {
                     let pos = record.pos() as u32;
                     let end_pos = cigar.end_pos() as u32;
 
-                    let overlap = {
+                    let overlap = { // BUG: SoftClips can be preceded by HardClips at read starts, followed by HardClips at read ends!
+                        // TODO: suggestion for removing the above BUG: consider this in the CigarStringView::end_pos calculation by adding in the include_softclips argument there, as well, and create an according function CigarStringView::start_pos
```

Comment:
Ok, I guess my explanation wasn't clear: The problem is the comparison to the first and last cigar operation, as SoftClips can at the beginning of the read be preceded by and at the end of the read followed by HardClips. I.e. if you have a CigarString that starts with `5H2S...`, the comparison `Cigar::SoftClip(l) = cigar[0]` will return `false` and you'll miss making the desired `pos`-adjustment for the `SoftClip` at the second Cigar position. Am I making sense? Or am I missing something that is going on in this code?

I guess, this could actually be avoided by one simple extra check here:
```
let pos = if let Cigar::SoftClip(l) = cigar[0] |
    ( Cigar::HardClip(_) = cigar[0] & Cigar::SoftClip(l) = cigar[1] ) {
```